vms/platformvm/state/stakers_crud_ops_test.go

```diff
@@ -0,0 +1,535 @@
+			}
+		})
+	}
+}
+
+func TestPendingDelegatorsCRUD(t *testing.T) {
+	subnetID := ids.GenerateTestID()
+
+	tests := []struct {
+		name         string
+		stakersStore func() (Stakers, error)
+	}{
+		{
+			name: "base state",
+			stakersStore: func() (Stakers, error) {
+				baseDBManager := manager.NewMemDB(version.Semantic1_0_0)
+				baseDB := versiondb.New(baseDBManager.Current().Database)
+				return buildChainState(baseDB, nil)
+			},
+		},
+		{
+			name: "diff",
+			stakersStore: func() (Stakers, error) {
+				diff, err := buildDiffOnTopOfBaseState([]ids.ID{subnetID})
+				return diff, err
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			store, err := tt.stakersStore()
+			require.NoError(err)
+
+			var (
+				now               = time.Now()
+				delegatorDuration = 30 * 24 * time.Hour
+
+				delegator = &Staker{
+					TxID:     ids.GenerateTestID(),
+					NodeID:   ids.GenerateTestNodeID(),
+					SubnetID: subnetID,
+					Weight:   uint64(2023),
+
+					StartTime: now,
+					NextTime:  now.Add(delegatorDuration),
+					EndTime:   now.Add(delegatorDuration),
+					Priority:  txs.PrimaryNetworkDelegatorBanffPendingPriority,
+				}
+			)
+
+			{
+				// no delegator before insertion
+
+				// query
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// it's fine deleting unknown delegator
+				store.DeletePendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// insert the delegator and show it can be found
+				store.PutPendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{delegator}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+
+			{
+				// delete the delegator and show it's not found anymore
+				store.DeletePendingDelegator(delegator)
+
+				// query
+				expectedResult := []*Staker{}
+				require.NoError(checkDelegatorsContent(store, pending, delegator.SubnetID, delegator.NodeID, expectedResult))
+
+				// range query
+				require.NoError(checkStakersContent(store, pending, expectedResult))
+			}
+		})
+	}
+}
+
+// [checkStakersContent] verifies whether store contains exactly the stakers specified in the list.
+// stakers order does not matter. stakers slice gets consumed while checking.
+func checkStakersContent(store Stakers, validatorsType uint8, expectedValidators []*Staker) error {
```

Comment:
Instead of having this return an error, it could take in a `require`, which personally I prefer, but up to you.